core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.spec.ts

```diff
@@ -0,0 +1,290 @@
+    let de: DebugElement;
+    let component: DotFavoritePageComponent;
+    let injector: TestBed;
+    let dialogRef: DynamicDialogRef;
+    let store: DotFavoritePageStore;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                DotFavoritePageComponent,
+                DotMessagePipe,
+                DotFormDialogMockComponent,
+                DotHtmlToImageMockComponent
+            ],
+            imports: [
+                FormsModule,
+                MultiSelectModule,
+                ReactiveFormsModule,
+                DotFieldValidationMessageModule,
+                HttpClientTestingModule
+            ],
+            providers: [
+                { provide: DotRouterService, useClass: MockDotRouterService },
+                { provide: CoreWebService, useClass: CoreWebServiceMock },
+                {
+                    provide: LoginService,
+                    useClass: LoginServiceMock
+                },
+                {
+                    provide: DotMessageService,
+                    useValue: messageServiceMock
+                },
+
+                {
+                    provide: DynamicDialogRef,
+                    useValue: {
+                        close: jasmine.createSpy()
+                    }
+                },
+                {
+                    provide: DynamicDialogConfig,
+                    useValue: {
+                        data: {
+                            page: {
+                                order: 1,
+                                pageState: mockRenderedPageState,
+                                pageRenderedHtml: '<p>test</p>'
+                            }
+                        }
+                    }
+                }
+            ]
+        }).compileComponents();
+        TestBed.overrideProvider(DotFavoritePageStore, { useValue: storeMock });
+        store = TestBed.inject(DotFavoritePageStore);
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(DotFavoritePageComponent);
+        de = fixture.debugElement;
+        component = fixture.componentInstance;
+        injector = getTestBed();
+
+        dialogRef = injector.inject(DynamicDialogRef);
+
+        fixture.detectChanges();
+    });
+
+    describe('HTML', () => {
+        it('should setup <form> class', () => {
+            const form = de.query(By.css('[data-testId="form"]'));
+            expect(form.classes['p-fluid']).toBe(true);
+        });
+
+        describe('fields', () => {
+            it('should setup thumbnail preview', () => {
+                const field = de.query(By.css('[data-testId="thumbnailField"]'));
+                const label = field.query(By.css('label'));
+                const webcomponent = field.query(By.css('dot-html-to-image'));
+
+                expect(field.classes['field']).toBe(true);
+
+                expect(label.classes['p-label-input-required']).toBe(true);
```

Comment:
Shouldn't we just check that what are passing correctly?

Reply:
no, that class is used for required star
![image](https://user-images.githubusercontent.com/37185433/181849241-ad41d1ee-fbd2-48e6-b50b-88b3eb534e81.png)